localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -368,6 +372,28 @@ void NDTScanMatcher::publish_diagnostic()
   diagnostics_pub_->publish(diag_msg);
 }
 
+void NDTScanMatcher::callback_timer()
+{
+  if (!is_activated_) {
+    return;
+  }
+  if (current_position_ == std::nullopt) {
+    RCLCPP_ERROR_STREAM_THROTTLE(
+      this->get_logger(), *this->get_clock(), 1000,
+      "Cannot find the reference position for map update. Please check if the EKF odometry is "
+      "provided to NDT.");
+    return;
+  }
+  if (!use_dynamic_map_loading_) {
+    return;
+  }
```

Comment:
It would be better to check `use_dynamic_map_loading_` first.
I don't think it is proper to give the above error message when the variable is false.

Reply:
It make super sense.
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/5915/commits/fbe915757393a5d0b876a11db64aa453c07fa7d8